runtime/near-vm/engine/src/universal/builder.rs

```diff
@@ -35,17 +41,26 @@ impl Universal {
         self
     }
 
+    /// Set the pool of reusable code memory
+    pub fn code_memory_pool(mut self, pool: super::LimitedMemoryPool) -> Self {
+        self.pool = Some(pool);
+        self
+    }
+
     /// Build the `UniversalEngine` for this configuration
     pub fn engine(self) -> UniversalEngine {
         let target = self.target.unwrap_or_default();
+        let pool = self.pool.unwrap_or_else(|| {
+            super::LimitedMemoryPool::new(1, 10 * rustix::param::page_size()).unwrap()
```

Comment:
And I am also a bit confused, what exactly happens here? Does it allocate a new memory pool wit a new `CodeMemory` in case we don't have a pool? Is it mostly for tests or something?